src/qibolab/platforms/abstract.py

```diff
@@ -38,45 +118,27 @@ def __init__(self, name, runcard):
             self.resonator_type = "3D"
         else:
             self.resonator_type = "2D"
-        self.hardware_avg = self.settings["settings"]["hardware_avg"]
```

Comment:
These cannot be moved from `init()` to `reload_settings()`, they need to be set in both places, perhaps with a routine that can be called from both functions.